packages/images/src/main.ts

```diff
@@ -0,0 +1,131 @@
+import type { Logger } from '@netlify/dev-utils'
+import { createIPX, createIPXWebServer, ipxFSStorage, ipxHttpStorage } from 'ipx'
+
+interface ImagesConfig {
+  remote_images: string[]
+}
+
+interface ImageHandlerOptions {
+  imagesConfig?: ImagesConfig
+  logger: Logger
+  originServerAddress?: string
+}
+
+export interface ImageMatch {
+  handle: () => Promise<Response>
+}
+
+const IMAGE_CDN_ENDPOINTS = ['/.netlify/images', '/.netlify/images/']
+
+export class ImageHandler {
+  #allowedRemoteUrlPatterns: RegExp[]
+  #logger: Logger
+  #originServerURL?: URL
+
+  constructor({ logger, imagesConfig, originServerAddress }: ImageHandlerOptions) {
+    this.#logger = logger
+    this.#allowedRemoteUrlPatterns = (imagesConfig?.remote_images ?? []).reduce<RegExp[]>((acc, stringPattern) => {
+      try {
+        acc.push(new RegExp(stringPattern))
+      } catch (maybeError) {
+        const error = maybeError instanceof Error ? maybeError : new Error(String(maybeError))
+        this.#logger.warn(`Malformed remote image pattern: "${stringPattern}": ${error.message}. Skipping it.`)
+      }
+      return acc
+    }, [])
+    this.#originServerURL = originServerAddress ? new URL(originServerAddress) : undefined
+  }
+
+  private generateIPXRequestURL(imageURL: URL, netlifyImageCdnParams: URLSearchParams): URL {
```

Comment:
heavily inspired by https://github.com/netlify/cli/blob/03fd72d88ac1fb598358807049bfa76707862fa0/src/lib/images/proxy.ts#L92 (and adjusted a bit to make our eslint happy)